.pre-commit-config.yaml

```diff
@@ -99,16 +99,41 @@ repos:
         entry: python ./.pre_commit/check-redactions.py
         always_run: true
         pass_filenames: false
-      - id: pmd
-        name: pmd
+      - id: pmd-client-java
+        name: pmd-client-java
         description: "Runs the PMD static code analyzer."
-        language: docker_image
-        entry: -v ./.pmd_cache:/opt/.pmd_cache --entrypoint .pre_commit/run-pmd.sh cimg/openjdk:11.0
+        language: script
+        entry: .pre_commit/run-pmd.sh --java-version 17 --ruleset-file client/java/pmd-openlineage.xml
         files: ^client/java/src/.*\.java$
         exclude: ".*test.*"
         require_serial: true
       - id: spec-changes
         name: Verify spec changes committed to website
         language: script
         files: ^spec/
-        entry: .pre_commit/run-spec-snapshot.sh
\ No newline at end of file
+        entry: .pre_commit/run-spec-snapshot.sh
+      - id: pmd-flink
```

Comment:
Would it be possible to have a single pmd ruleset and verify all the projects with a single run?

Reply:
It would be most convenient I would say.
However, currently Java projects have different rulesets, I'm not sure what's the best way to establish common rulesets.